### PR TITLE
Correct bugs

### PR DESCRIPTION
**Pull request by tbrezot**

### CODE REVIEW

File: src/big_int.rs [hunk 1]
```diff
@@ -291,24 +285,32 @@ impl BigInt {
...
+        }
     }
 
     /// Reduce `self` modulo the given `u64`.
-    pub fn reduce_ui(&self, modulus: u64) -> u64 {
-        u64::try_from(&self.reduce(&BigInt::from(modulus)))
-            .expect("Reducing a `BigInt` by a `u64` should return a 64-bit sized `BigInt`")
+    pub fn reduce_ui(&self, modulus: u64) -> Result<u64, BigIntError> {
+        Ok(u64::try_from(&self.reduce(&BigInt::from(modulus))?)
+            .expect("Reducing a `BigInt` by a `u64` should return a 64-bit sized `BigInt`"))
```
Comment (ThibsG) [inline src/big_int.rs]: now this method returns `Result`, do you think we can get rid of `.expect()`?

File: src/big_int.rs [hunk 2]
```diff
@@ -323,31 +325,34 @@ impl BigInt {
     /// This function will panic if it is called on a negative number.
     ///
     /// - `modulus` : list of primes `p_i`
-    pub fn to_rns(&self, modulus: &[u64]) -> RNSRepresentation {
+    pub fn to_rns(&self, modulus: &[u64]) -> Result<RNSRepresentation, BigIntError> {
         assert!(
             self >= &BigInt::zero(),
             "negative BigInt to RNSRepresentation convversion is not yet supported!"
```
Comment (ThibsG) [inline src/big_int.rs]:
```suggestion
            "negative BigInt to RNSRepresentation conversion is not yet supported!"
```

Comment (tbrezot) [reply src/big_int.rs]: Done

File: src/big_int.rs (hunk 1 above)
Comment (tbrezot) [reply src/big_int.rs]: Done

**Pull request merged**